### PR TITLE
feat: add known error signals

### LINKED ISSUE #10: Add known error signals
We have some known error signals [here](https://github.com/FuelLabs/sway/blob/master/sway-lib-std/src/error_signals.sw) and features like https://github.com/FuelLabs/sway/pull/3999 needs to access this information. As discussed [here](https://github.com/FuelLabs/sway/issues/3950#issuecomment-1411333835) it makes sense to host known error signals in this repo as this is shared between `forc` and the sdk.

### PR DESCRIPTION
**Pull request by kayagokalp**
closes #10.

### CODE REVIEW

File: src/error_codes.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            Self::SendMessage
+        } else if revert_code == FAILED_ASSERT_EQ_SIGNAL {
+            Self::AssertEq
+        } else if revert_code == FAILED_ASSERT_SIGNAL {
+            Self::Assert
+        } else {
+            Self::Unknown
+        }
+    }
+}
```
Comment (sdankel) [inline src/error_codes.rs:50]: A ```pub fn to_revert_code(&self) -> u64``` could be useful too.

Comment (kayagokalp) [reply src/error_codes.rs:64]: Good idea! Thanks

**Review by digorithm — DISMISSED** [review]
LGTM.

@hal3e, we should remove these consts from the SDK and pull them from here once this is released. Thoughts?

File: src/error_codes.rs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    #[error("Failing call to `std::token::transfer_to_address`")]
+    TransferToAddress,
+    #[error("Failing call to `std::message::send_message`")]
+    SendMessage,
+    #[error("Failing call to `std::assert::assert_eq`")]
+    AssertEq,
+    #[error("Failing call to `std::assert::assert`")]
+    Assert,
+    #[error("Unknown error signal")]
+    Unknown,
```
Comment (mitchmindtree) [inline src/error_codes.rs:31]: I would maybe remove this `Unknown` variant in favour of changing `from_revert_code` to something like:

```rust
pub fn try_from_revert_code(revert_code: u64) -> Result<Self, UnknownSignal>
```

**Review by mitchmindtree — DISMISSED** [review]
One optional suggestion, otherwise lgtm!

**Review by sdankel — APPROVED** [review]

**Comment by hal3e** [comment]
> LGTM.
> 
> @hal3e, we should remove these consts from the SDK and pull them from here once this is released. Thoughts?

Yes. I will add an issue. (we will need a new release of this repo)

Additionally, we should think about renaming this repo because we are not limiting it to abi types.

**Pull request merged**